website/content/docs/intro/use-cases/dynamic-config-vault-dynamic-secrets.mdx

```diff
@@ -0,0 +1,269 @@
+
+```shell
+$ waypoint config source-set -type=vault -config=addr=https://vault.service.consul:8200 -config=token=<VAULT_TOKEN>
+Configuration set for dynamic source "vault"!
+```
+
+We can double-check that the token and address are what we expect as well:
+
+```shell
+$ waypoint config source-get -type=vault
+  KEY  |           VALUE
+--------+----------------------------
+  addr  | https://vault.service.consul:8200
+  token | <VAULT_TOKEN>
+```
+
+At this point, we aren’t actually using the config source yet, but we can get that
+ready by adding it to our waypoint.hcl file, and committing it to the git repository
+for the project.
+
+```hcl
+project = “waypoint-dynamic-config-vault”
+
+app “my-app” {
+  build {
+    use “pack” {}
+
+    registry {
+      use “docker” {
+        image = var.image_name
+        tag   = var.image_tag
+        auth {
+          username = var.username
+          password = var.password
+        }
+      }
+    }
+  }
+
+  deploy {
+    use “nomad” {
+      // The application code is written such that the app will listen on port 80,
+      // so Nomad should configure the job to listen on this port.
+      service_port = 80
+    }
+  }
+
+  config {
+    env = {
+      // These environment variables use dynamic configuration from Vault's dynamic
+      // database secrets engine
+      USERNAME = dynamic(“vault”, {
+        path = “database/creds/my-role”
+        key = “username”
+      })
+
+      PASSWORD = dynamic(“vault”, {
+        path = “database/creds/my-role”
+        key = “password”
+      })
+
+      HOST   = var.postgres_ip
+      PORT   = var.postgres_port
+      DBNAME = var.postgres_dbname
+    }
+  }
+}
+
+variable "postgres_ip" {
+  description = "IP address at which the Postgres DB server is listening."
+  type        = string
+}
+
+variable "postgres_port" {
+  description = "Port on which the Postgres DB server is listening."
+  type        = number
+  default     = 5432
+}
+
+variable "postgres_dbname" {
+  description = "Name of the Postgres DB to access."
+  type        = string
+  default     = “postgres”
+}
+
+variable "image_name" {
+  decription = "Name of the Docker image for the app built by Waypoint. This will be used with the tag to push an image to a Docker repository."
+  type = string
+}
+
+variable "image_tag" {
+  decription = "Tag of the Docker image for the app built by Waypoint. This will be used with the name to push an image to a Docker repository."
+  type = string
+}
+
+variable "registry_username" {
+  description = "Username of user with push access to your Docker registry."
+  type      = string
+  sensitive = true
+}
+
+variable "registry_password" {
+  description = "Password for the configured user with push access to your Docker registry."
+  type      = string
+  sensitive = true
+}
+```
+
+In the configuration above, we’ve informed Waypoint that we want the Waypoint Entrypoint,
```

Comment:
Maybe turn `Waypoint Entrypoint` into a URL to link to what that is in the docs?

Reply:
I wonder if we can avoiding introducing the concept of the CEB altogether here. It’s pretty advanced stuff and may leave the reader feeling overloaded with concepts. Loose suggestion:

```markdown
In the configuration above, we’ve informed Waypoint that we want to get a
secret from Vault at the path `database/creds/my-role` and make it available in
our app’s container. To learn more about how Waypoint does this, see the
[Waypoint Entrypoint docs](https://developer.hashicorp.com/waypoint/docs/entrypoint).
```